src/DotVVM.Framework/ResourceManagement/ResourceManager.cs

```diff
@@ -114,6 +114,16 @@ public void AddRequiredScriptFile(string name, string url, params string[] depen
             });
         }
 
+        /// <summary>
+        /// Adds the required inline script.
+        /// </summary>
+        public void AddRequiredInlineScript(string name, string code, params string[] dependentResourceNames)
```

Comment:
Ah....
I was looking whether we have something exactly like AddStartupScript, but because of its name I didn't event looked at this method.

The name is really terrible. I don't know if it's worth breaking change, but I would suggest to mark this method as obsolete and replace (rename) it.

Reply:
This is probably used at hundreds of places, I guess we will never remove the method. We could however do the graceful rename, but I'm not sure if that's the right think to do.

Now, I'm thinking that maybe it's just better to have a generic `AddRequiredResource(optionla string name, IResource)`. It's not that noisy and might be much more discoverable.

Reply:
Ok, turns out that although we can add resources that are not registered globally, it's private...

Reply:
@tomasherceg Can we basically just make `AddRequiredResourceCore` public (I'd rename it to `AddRequiredResource`)? I don't see a problem with and it seems cleaner than having IDK how many helper methods for each resource type...

Reply:
OK, cant we than just make AddRequiredResourceCore public? I would also rename it. It's private so it should not break anyone.

Reply:
I think we can, would leave it here for some time, maybe we'll figure out that it's a bad idea actually :P But I always thought that's it's actually possible...

Reply:
ok I'll modify this PR to just make the AddRequiredResourceCore public for now.
What about renaming it? I would make it just another AddRequiredResource overload.

Reply:
>  I would make it just another AddRequiredResource overload.

yes, I also like this option.